docs/dev_utils.md

```diff
@@ -0,0 +1,342 @@
+### Utilities 
```

Comment:
I'm not the biggest fan of how this is formatted, let's change that while we have the chance